### PR TITLE
Tweak insertOnce tests and fix reviews tests.

### LINKED ISSUE #2993: investigate why is WP 5.4 + GB E2E test is failing and remove the skip
In https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2990, we added an `allow_fail` for WP 5.4 + GB E2E test because it kept failing for us on main, the reasons are not clear yet, but it could be an API change.

### PR DESCRIPTION
**Pull request by senadir**
This PR does a couple of things:

- Fixes some fragile Reviews tests that relied on all products having a review, now we will select products or categories that actually have reviews (using fixture data), it also introduces a new helper function `getElementByText` that gets an element by class name and content (static or regex).

- Refactor our `can only be inserted once` tests, we had problems with `insertBlock` helper function because it waited for the block to actually insert before moving forward, causing a timeout, we now search for the block and check if the button is disabled or not.

### New utilities

- `async closeInserter()` is now a global utility that you can use, it will close the inserter after failed inserts.
- `async getElementByText( selector<String>, pattern<String> )` that will return an element using a selector and a pattern that can be a regular expression.

fixes #2993, I removed the allow fail declaration. 
### How to confirm changes
Travis should pass for this test.

### CODE REVIEW

File: tests/e2e-tests/specs/backend/reviews-by-category.test.js [hunk 1]
```diff
@@ -32,7 +40,11 @@ describe( `${ block.name } Block`, () => {
 		await page.waitForSelector(
 			`${ block.class } .woocommerce-search-list__item`
 		);
-		await page.click( `${ block.class } .woocommerce-search-list__item` );
+		const categoryWithReviews = await findElementWithText(
+			`.woocommerce-search-list__item`,
+			fixtureProducts()[ 0 ].categories[ 0 ]
+		);
+		await categoryWithReviews.click();
```
Comment (Aljullu) [inline tests/e2e-tests/specs/backend/reviews-by-category.test.js:47]: I don't know why this test is failing for me here. I tried debugging it with `npm run test:e2e-dev` and I saw only the 'Uncategorized' item is shown in the selector:

imatge

I can keep investigating what's going on, but if somebody has any idea, that would be appreciated. :slightly_smiling_face:

**Review by Aljullu — COMMENTED** [review]
One of the tests is failing for me (it might be something wrong with my local install, considering that it's passing in Travis), so I'm unable to give an approval to the PR, but I took a look and overall code looks good.

File: tests/e2e-tests/specs/backend/reviews-by-category.test.js (hunk 1 above)
Comment (senadir) [reply tests/e2e-tests/specs/backend/reviews-by-category.test.js:47]: It seems you don't have a category named `Music` that should be automatically created as part of the fixtures, I'm not really sure why your instance fails to insert it.
https://github.com/woocommerce/woocommerce-gutenberg-products-block//blob/116e962aa4b1cd63252a18988506d911a5da2814/tests/e2e-tests/config/setup.js#L36

File: tests/e2e-tests/specs/backend/cart.test.js [hunk 2]
```diff
@@ -35,9 +32,15 @@ describe( `${ block.name } Block`, () => {
...
 	it( 'can only be inserted once', async () => {
-		await insertBlock( block.name );
+		await searchForBlock( block.name );
+		const disabledInsertButton = await findElementWithText(
+			'button.editor-block-list-item-woocommerce-cart',
+			block.name
+		);
+		expect(
+			await disabledInsertButton.evaluate( ( button ) => button.disabled )
+		).toBe( true );
```
[the remark this answers is not in the document]
Comment (senadir) [reply tests/e2e-tests/specs/backend/cart.test.js:42]: Yeah I thought of that.

**Comment by nerrad** [comment]
FYI: I ran the e2e tests in this branch and they all passed for me locally.

**Comment by nerrad** [comment]
@senadir this pull has been open for a while, besides a rebase, what else is needed to see this through to merge? Will it potentially improve the intermittent test fails we've been seeing?

**Comment by senadir** [comment]
After several attempts to rebase this, I'm going to simply close and start over.

**Pull request closed without merging**